File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Projects/fiscalyearproperties.rb
module Asposetasksjava
  module FiscalYearProperties
    def initialize()
        # Reading Fiscal Year Properties
        get_fiscal_year_properties()

        # Writing Fiscal Year Properties 
        set_fiscal_year_properties()
    end

    def get_fiscal_year_properties()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        prj = Rjb::import('com.aspose.tasks.Prj')

        # Display fiscal year properties
        puts "Fiscal Year Start Date : " + project.get(prj.FY_START_DATE).toString()
        puts "Fiscal Year Numbering : " + project.get(prj.FISCAL_YEAR_START).toString()
    end

    def set_fiscal_year_properties()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new

        prj = Rjb::import('com.aspose.tasks.Prj')

        # Set fiscal year properties
        project.set(prj.FY_START_DATE, Rjb::import('com.aspose.tasks.Month').July)
        project.set(prj.FISCAL_YEAR_START, Rjb::import('com.aspose.tasks.NullableBool').new(true))

        project.save(data_dir + "fiscal_year_properties.xml", Rjb::import('com.aspose.tasks.SaveFileFormat').XML)

        puts "Set fiscal year properties, please check the output file."
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithProjects/FiscalYearProperties.php
<?php
namespace Aspose\Tasks\WorkingWithProjects;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\Prj as Prj;
use com\aspose\tasks\Month as Month;
use com\aspose\tasks\NullableBool as NullableBool;
use com\aspose\tasks\SaveFileFormat as SaveFileFormat;

class FiscalYearProperties{

    public static function run($dataDir=null){

        # Reading Fiscal Year Properties
        FiscalYearProperties::get_fiscal_year_properties($dataDir);

        # Writing Fiscal Year Properties
        FiscalYearProperties::set_fiscal_year_properties($dataDir);
    }

    public static function get_fiscal_year_properties($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        $prj = new Prj();

        # Display fiscal year properties
        print "Fiscal Year Start Date : " . (string)$project->get($prj->FY_START_DATE).PHP_EOL;//.toString();
        print "Fiscal Year Numbering : " . (string)$project->get($prj->FISCAL_YEAR_START).PHP_EOL;//.toString()
    }

    public static function set_fiscal_year_properties($dataDir=null){

        # Instantiate project object
        $project = new Project();

        $prj = new Prj();

        # Set fiscal year properties
        $month=new Month();
        $project->set($prj->FY_START_DATE, $month->July);
        $project->set($prj->FISCAL_YEAR_START, new NullableBool(true));

        $saveFileFormat=new SaveFileFormat();

        $project->save($dataDir . "fiscal_year_properties.xml", $saveFileFormat->XML);
        print "Set fiscal year properties, please check the output file.".PHP_EOL;
    }
}
?>